### PR TITLE
Lua: SMTP support; Addresses feature ticket #1775; v5

### PR DESCRIPTION
**Pull request by tobiass1**
Implemented support for matching on SMTP traffic from lua scripts.
This enables availibility to MIME fields, 
MailFrom attribute, recipients list.

Removed the functionality attachment information extraction introduced in previous revisions as this did not work as intended. 
Also fixed changes suggested in #2135 
 
Redmine ticket: https://redmine.openinfosecfoundation.org/issues/1775

### CODE REVIEW

File: src/util-lua-smtp.c [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    }
+
+    /* Create a new table in luastate for rcpt list */
+    lua_newtable(luastate);
+    /* rcpt var for iterator */
+    int u = 1;
+    SMTPString *rcpt;
+
+    TAILQ_FOREACH(rcpt, &smtp_tx->rcpt_to_list, next) {
+        LuaPushStringBuffer(luastate, rcpt->str, rcpt->len);
```
Comment (inliniac) [inline src/util-lua-smtp.c]: Here the table row is : buffer, int

File: src/util-lua-smtp.c [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    if(field->name == NULL || field->name_len == 0) {
+        return LuaCallbackError(luastate, "Error: field has no name");
+    }
+    /* Counter of MIME fields found */
+    int num = 1;
+    /* loop trough the list of mimeFields, printing each name found */
+    lua_newtable(luastate);
+    while (field != NULL) {
+        if(field->name != NULL && field->name_len != 0) {
+            lua_pushinteger(luastate,num++);
```
Comment (inliniac) [inline src/util-lua-smtp.c]: here the table row is int, buffer.

**Comment by inliniac** [comment]
Is there a specific reason for the different ways the tables are setup?

**Comment by tobiass1** [comment]
@inliniac - No specific reason. Differently formatted table-output to lua will be the only affect.
Just us being inconsistent.

**Comment by inliniac** [comment]
On 07-07-16 09:12, tobiass1 wrote:
> @inliniac <https://github.com/inliniac> - No specific reason.
> Differently formatted table-output to lua will be the only affect.
> Just us being inconsistent.

OK. I'm changing SMTPGetRcptList to use the position as the key instead
of value.

Thanks!

**Comment by inliniac** [comment]
Merged through #2168, thanks guys!

**Comment by pevma** [comment]
@tobiass1 - much appreciated if  you could please include some basic usage examples on our wiki so that users can actually refer to how it is used and contribute back.

**Comment by pevma** [comment]
@tobiass1 - friendly reminder :)

**Pull request closed without merging**